<launch>

<node pkg="topic_tools" type="throttle" name="base_scan_throttle" args="messages base_scan 1.0 base_scan_throttled" machine="c2" output="screen"/>

<!--
<node pkg="laser_filters" type="scan_to_cloud_filter_chain" respawn="true" machine="c2" name="tilt_shadow_filter">
  <remap from="scan" to="tilt_scan" />
  <remap from="cloud_filtered" to="tilt_scan_marking" />
  <param name="target_frame" value="base_footprint" />
  <param name="high_fidelity" value="true" />
  <rosparam command="load" file="$(find pr2_navigation_perception)/config/shadow_filter.yaml" />
  <rosparam command="load" file="$(find pr2_navigation_perception)/config/point_cloud_footprint_filter.yaml" />
</node>
-->

<!-- Filter for base laser shadowing/veiling -->
<node pkg="laser_filters" type="scan_to_cloud_filter_chain" respawn="true" machine="c2" name="base_shadow_filter" >
  <remap from="scan" to="base_scan" />
  <remap from="cloud_filtered" to="base_scan_marking" />
  <param name="target_frame" value="base_footprint" />
  <param name="high_fidelity" value="true" />
  <rosparam command="load" file="$(find pr2_navigation_perception)/config/shadow_filter.yaml" />
  <rosparam command="load" file="$(find pr2_navigation_perception)/config/point_cloud_footprint_filter.yaml" />
</node>
</launch>
